### ISSUE TITLE
False positive: literals in unions

### ISSUE THREAD

**Issue by Tinche** [python/mypy#11414]
**Bug Report**

Mypy reports an error where it shouldn't. Sample code included below.

**To Reproduce**

```
from typing import Literal, Union


def test() -> Union[tuple[Literal[1], str], tuple[Literal[2], int]]:
    return 1, ""
```

**Expected Behavior**

The code should successfully type-check.

**Actual Behavior**

Mypy does not successfully type check the code, reporting:

```
error: Incompatible return value type (got "Tuple[int, str]", expected "Union[Tuple[Literal[1], str], Tuple[Literal[2], int]]")
```

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none (defaults)
- Python version used: 3.9.7
- Operating system and version: Ubuntu 21.10

**Comment by sobolevn** [comment]
Please, check if https://github.com/python/mypy/pull/11236/files solves your issue!

**Comment by Tinche** [comment]
@sobolevn It does! Thanks, closing this.
